src/actions/index.js

```diff
@@ -163,15 +184,178 @@ export const devicesDelete = (deviceId) => {
 
 export const devicesCreate = (device, type) => {
   return (dispatch) => {
+    var newDevice;
     dispatch(loading(true))
     agile.deviceManager.create(device, type)
-    .then((newDevice) => {
-      dispatch(action('DEVICES_CREATE', newDevice));
-      dispatch(loading(false));
-    })
-    .catch(err => {
-      errorHandle(err, dispatch)
+      .then((d) => {
+        newDevice = d;
+        return agile.idm.entity.get(d.deviceId, DEVICE_TYPE);
+      })
+      .then((entity) => {
+        return Promise.resolve(entity);
+      })
+      .catch(err => {
```

Comment:
This should be `then` right?

Reply:
I don't see the then missing. I don't know if this is what you meant, but there is an issue. If the entity was in IDM, we return the device info from IDM and not from the device manager... So this needs to be fixed, but otherwise, if the device is not in IDM we try to create it in the catch.